x-pack/plugins/maps/server/maps_telemetry/maps_usage_collector.js

```diff
@@ -13,37 +13,58 @@ export function initTelemetryCollection(server) {
   registerMapsUsageCollector(server);
 }
 
-export function buildCollectorObj(server) {
-  return {
-    type: 'maps',
-    fetch: async () => {
-      let docs;
-      try {
-        ({ docs } = await server.taskManager.fetch({
-          query: {
-            bool: {
-              filter: {
-                term: {
-                  _id: TASK_ID
-                }
-              }
+async function isTaskManagerReady(server) {
+  const result = await fetch(server);
+  return result !== null;
+}
+
+async function fetch(server) {
+  let docs;
+  try {
+    ({ docs } = await server.taskManager.fetch({
+      query: {
+        bool: {
+          filter: {
+            term: {
+              _id: TASK_ID
             }
           }
-        }));
-      } catch (err) {
-        const errMessage = err && err.message ? err.message : err.toString();
-        /*
-         * The usage service WILL to try to fetch from this collector before the task manager has been initialized, because the task manager
-         * has to wait for all plugins to initialize first.
-         * It's fine to ignore it as next time around it will be initialized (or it will throw a different type of error)
-         */
-        if (errMessage.indexOf('NotInitialized') >= 0) {
-          docs = {};
-        } else {
-          throw err;
         }
       }
+    }));
+  } catch (err) {
+    const errMessage = err && err.message ? err.message : err.toString();
```

Comment:
You could shorten this to `err.message ? err.message : err.toString();`

Reply:
I didn't originally write this, but happy to update it

Reply:
Nope, I did :) . Not critical either way. Thanks though!